### PR TITLE
Fixes #1108: Temporary Fix for Deprecated Qt functions

### LINKED ISSUE #1108: [Source]: Deprecated Qt functions fail build on Arch
### Operating System Version

Arch Linux x86_64

### CPU

Intel(R) Core(TM) i5-3470 (4) @ 3.60 GHz

### GPU

NVIDIA Quadro T400 Mobile

### NDI Devices

_No response_

### OBS Version

30.2.1

### OBS Installation Method

pacman

### DistroAV Version

4.14.1

### DistroAV Installation Method

AUR(Arch User Repo)

### [Extra] Installation Steps

_No response_

### OBS Log [URL]

https://obsproject.com/logs/K9N1635h1ZBhhHTM

### NDI Version

4.14.1

### Describe the bug

Got build failure:

image

There are two deprecated functions inside **`src/forms/obsndi-update.cpp`**-> 

```
1. utcDateTime.setTimeSpec(Qt::UTC);
2. &QCheckBox::stateChanged
```

### Steps to reproduce the problem

Just installed and build the package with AUR helper "trizen".
```
git clone https://aur.archlinux.org/trizen && makepkg -cCsi -D trizen/ && rm -rf trizen/
trizen -S ndi-sdk 
trizen -S obs-ndi 
```

### Expected behavior

_No response_

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AniketBose369**

### CODE REVIEW

**Comment by paulpv** [comment]
@AniketBose369 Please resolve the conflicts.
Why do you call these fixes "Temporary"?
Just remove the old code you commented out.

File: src/forms/obsndi-update.cpp [hunk 1]
```diff
@@ -174,7 +176,8 @@ class ObsNdiUpdate : public QDialog {
 
 		auto utcDateTime = QDateTime::fromString(
 			pluginUpdateInfo.releaseDate, Qt::ISODate);
-		utcDateTime.setTimeSpec(Qt::UTC);
+		//utcDateTime.setTimeSpec(Qt::UTC); fix it in issue-1108
```
Comment (paulpv) [inline src/forms/obsndi-update.cpp:179]: remove this commented out line

File: src/forms/obsndi-update.cpp [hunk 2]
```diff
@@ -190,7 +193,8 @@ class ObsNdiUpdate : public QDialog {
 		ui->checkBoxAutoCheckForUpdates->setChecked(
 			config->AutoCheckForUpdates());
 		connect(ui->checkBoxAutoCheckForUpdates,
-			&QCheckBox::stateChanged, this, [](int state) {
+			//&QCheckBox::stateChanged, fix it in issue-1108
```
Comment (paulpv) [inline src/forms/obsndi-update.cpp:196]: remove this commented out line

File: src/forms/obsndi-update.cpp [hunk 3]
```diff
@@ -49,6 +49,8 @@ Inspiration(s):
 #include <QTimer>
 #include <QUrlQuery>
 
+#include <QTimeZone>
```
Comment (paulpv) [inline src/forms/obsndi-update.cpp:52]: group with above #includes sorted alphabetically

**Comment by AniketBose369** [comment]
Ok

**Comment by AniketBose369** [comment]
Done

**Comment by paulpv** [comment]
@AniketBose369 Please fix the conflicts
image

File: src/forms/update.cpp [hunk 4]
```diff
@@ -183,8 +184,13 @@ class PluginUpdate : public QDialog {
 		ui->checkBoxAutoCheckForUpdates->setChecked(
 			config->AutoCheckForUpdates());
 		connect(ui->checkBoxAutoCheckForUpdates,
+<<<<<<< HEAD:src/forms/obsndi-update.cpp
```
Comment (paulpv) [inline src/forms/update.cpp:187]: This is not how to resolve a code conflict

**Review by paulpv — CHANGES_REQUESTED** [review]

**Comment by BitRate27** [comment]
It seems obs-studio reverted the move of Qt to 6.7 on Sep 12 ( https://github.com/obsproject/obs-deps/commit/a0d86669aeb254a8778df535df0a9767ada30625 ) and went back to 6.6.3. This means the new checkStateChanged signal is not in the Qt from obs-studio rn.

**Comment by AniketBose369** [comment]
Actually I compiled and run DistroAV(basically linked the libndi.so.6 with
5) with the changes I made and the latest obs ran fine.

**Comment by BitRate27** [comment]
Not exactly sure why that worked, but until they move to 6.7, we have to hold off as we won't be able to build. Since they reverted back, there must have been a major issue with 6.7, so we should wait until obs-studio settles on a new version. We can even look at this in Qt 6.8 as this one API isn't removed until 6.9.

**Comment by paulpv** [comment]
@AniketBose369 @BitRate27 Shall we close this for now?
Once OBS officially supports Qt 6.7/8 we will update ours and this warning error should be obvious and then fixed.

**Comment by AniketBose369** [comment]
Yeah that seems the right choice

**Comment by Trouffman** [comment]
PR closed as this is "too early".

Revise / re-submit PR when : 
* OBS integrate QT 6.7+

To-do :
*  Update minimum required OBS version for DistroAV to support the new minimum requirement.

**Comment by Trouffman** [comment]
@AniketBose369 a fix for you need has been merged to master. see #1151

**Pull request closed without merging**